sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/SparkUtils.scala

```diff
@@ -25,6 +28,26 @@ private object SparkUtils {
     Runtime.getRuntime.availableProcessors
   }
 
+  def safeOpenConnectionInitCaches(container: CosmosAsyncContainer, logger: ILogger): Unit = {
```

Comment:
nit: This could just reuse the other method calling `logger.logWarning` as handler, to reduce code duplication

Reply:
thanks. good suggestion. addressed.